contracts/staking/test/end_to_end_tests/deployment.ts

```diff
@@ -0,0 +1,431 @@
+                );
+                await registerAssetProxyAndAssertSuccessAsync(
+                    multiAssetProxy,
+                    erc1155Proxy.address,
+                    constants.ERC1155_PROXY_ID,
+                );
+                await registerAssetProxyAndAssertSuccessAsync(
+                    multiAssetProxy,
+                    staticCallProxy.address,
+                    constants.STATIC_CALL_PROXY_ID,
+                );
+            });
+
+            it('should successfully add the exchange as an authority in the appropriate asset proxies', async () => {
+                // Authorize the exchange in all of the asset proxies, except for the static call proxy.
+                await authorizeAddressAndAssertSuccessAsync(erc20Proxy, exchange.address);
+                await authorizeAddressAndAssertSuccessAsync(erc721Proxy, exchange.address);
+                await authorizeAddressAndAssertSuccessAsync(erc1155Proxy, exchange.address);
+                await authorizeAddressAndAssertSuccessAsync(multiAssetProxy, exchange.address);
+            });
+
+            it('should successfully add the multi asset proxy as an authority in the appropriate asset proxies', async () => {
+                // Authorize the multi-asset proxy in the token asset proxies.
+                await authorizeAddressAndAssertSuccessAsync(erc20Proxy, multiAssetProxy.address);
+                await authorizeAddressAndAssertSuccessAsync(erc721Proxy, multiAssetProxy.address);
+                await authorizeAddressAndAssertSuccessAsync(erc1155Proxy, multiAssetProxy.address);
+            });
+        });
+
+        describe('staking specific', () => {
+            it('should have properly configured the staking proxy with the logic contract and read-only proxy', async () => {
+                // Ensure that the registered read-only proxy is correct.
+                const readOnlyProxyAddres = await stakingProxy.readOnlyProxy.callAsync();
+                expect(readOnlyProxyAddres).to.be.eq(readOnlyProxy.address);
+
+                // Ensure that the registered read-only proxy callee is correct.
+                const readOnlyProxyCalleeAddres = await stakingProxy.readOnlyProxyCallee.callAsync();
+                expect(readOnlyProxyCalleeAddres).to.be.eq(staking.address);
+
+                // Ensure that the registered staking contract is correct.
+                const stakingAddress = await stakingProxy.stakingContract.callAsync();
+                expect(stakingAddress).to.be.eq(staking.address);
+            });
+
+            it('should have initialized the correct parameters in the staking proxy', async () => {
+                // Ensure that the correct parameters were set.
+                const params = await stakingWrapper.getParams.callAsync();
+                expect(params.length).to.be.eq(6);
+                expect(params[0]).bignumber.to.be.eq(new BigNumber(864000)); // epochDurationInSeconds
```

Comment:
Optional nit:
I _think_ we can do `expect(params).to.deep.equal(expectedParams)` where `expectedParams` is an array of `BigNumber`s, since `BigNumbers` are just objects and will be correctly handled by deep equality